src/main/scala/com/fulcrumgenomics/util/miseq/Sample.scala

```diff
@@ -71,5 +73,26 @@ class Sample(val sampleOrdinal: Int,
       case Some(str) => Some(str.trim)
     }
   }
+
+  /** Sets the sample barcode based on the read structure.
+    *
+    * @param sampleBarcodeReadStructures a read structure per read (ex. i7 and i5) that contains only sample barcode segments.
+    */
+  def setSampleBarcode(sampleBarcodeReadStructures: Seq[ReadStructure]): this.type = {
+    import com.fulcrumgenomics.util.{SampleBarcode => ReadSegmentSampleBarcode}
+    if (sampleBarcodeReadStructures.exists(rs => !rs.forall { case s: ReadSegmentSampleBarcode => true; case _ => false } )) {
```

Comment:
If you're going to bother doing this check here ... why not just allow any read structure in and do the filtering yourself?